docs/design/datacontracts/ExecutionManager.md

```diff
@@ -0,0 +1,69 @@
+# Contract ExecutionManager
+
+This contract is for mapping a PC address to information about the
+managed method corresponding to that address.
+
+
+## APIs of contract
+
+**TODO**
+
+## Version 1
+
+**TODO** Methods
+
+### NibbleMap
+
+Version 1 of this contract depends on a "nibble map" data structure
+that allows mapping of a code address in a contiguous subsection of
+the address space to the pointer to the start of that a code sequence.
+It takes advantage of the fact that the code starts are aligned and
+are spaced apart to represent their addresses as a 4-bit nibble value.
+
+Given a contiguous region of memory in which we lay out a collection of non-overlapping code blocks that are
+not too small (so that two adjacent ones aren't too close together) and  where the start of each code block is preceeded by a code header aligned on some power of 2,
+we can break up the whole memory space into buckets of a fixed size (32-bytes in the current implementation), where
+each bucket either has a code block header or not.
+Thinking of each code block header address as a hex number, we can view it as: `[index, offset, zeros]`
+where each index gives us a bucket and the offset gives us the position of the header within the bucket.
+We encode each offset into a 4-bit nibble, reserving the special value 0 to mark the places in the map where a method doesn't start.
+
+To find the start of a method given an address we first convert it into a bucket index (giving the map unit)
+and an offset which we can then turn into the index of the nibble that covers that address.
+If the nibble is non-zero, we have the start of a method and it is near the given address.
+If the nibble is zero, we have to search backward first through the current map unit, and then through previous map
+units until we find a non-zero nibble.
+
+For example (all code addresses are relative to some unspecified base):
+
+Suppose there is code starting at address 304 (0x130)
+
+* Then the map index will be 304 / 32 = 9 and the byte offset will be 304 % 32 = 16
+* Because addresses are 4-byte aligned, the nibble value will be 1 + 16 / 4 = 5  (we reserve 0 to mean no method).
+* So the map unit containing index 9 will contain the value 0x5 << 22 (the map index 2 means we want the second nibble in the second map unit, and we number the nibbles starting from the most significant) , or 0x1400000
```

Comment:
minor nit: could pad `0x1400000` to `0x01400000` to match `0x00000005` down below.